### PR TITLE
Refactor tests using async/await and applying standardjs

### PR DESCRIPTION
**Pull request by maeb**
I've started refactoring the tests.

- [x] accessing_reql.js
- [x] administration.js
- [x] aggregation.js
- [x] backtrace.js
- [x] client-backtrace.js
- [x] control-structures.js
- [x] coverage.js
- [x] cursor.js
- [x] dates-and-times.js
- [x] datum.js
- [x] dequeue.js
- [x] document-manipulation.js
- [x] error.js
- [x] extra.js
- [x] geo.js
- [x] joins.js
- [x] manipulating-databases.js
- [x] manipulating-tables.js
- [x] math-and-logic.js
- [x] multiple-require.js
- [x] nodeify.js
- [x] pool_legacy.js
- [x] selecting-data.js
- [x] stable.js
- [x] stream.js
- [x] string-manipulation.js
- [x] transformations.js
- [x] transform-stream.js
- [x] writable-stream.js
- [x] writing-data.js

### CODE REVIEW

**Review by mxstbr — APPROVED** [review]
Yesss this is so much better!

File: test/cursor.js [hunk 1]
```diff
@@ -1,877 +1,700 @@
...
-
-    done();
-  }
-  catch(e) {
-    done(e);
-  }
-})
+  })
+
+  it('Import with cursor as default', async () => {
```
Comment (maeb) [inline test/cursor.js:529]: This test case causes the test to hang after completion. Newer versions of Mocha hangs on unresolved promises so it's a possible bug.

**Comment by thelinuxlich** [comment]
great work @maeb !

**Comment by maeb** [comment]
All tests passing without hanging. There might still be some timing issues that need to be sorted out that can maybe alleviated by specifying `--retries 2` parameter to run script.

**Comment by thelinuxlich** [comment]
Thanks @maeb nice work!

**Pull request merged**